### PR TITLE
feat: Add IDV attempt events

### PR DESCRIPTION
**Pull request by ilee2u**
This PR is a draft for now while my team looks over this.

**Description:** Adding events for status changes to the new VerificationAttempt model planned for edx-platform. 

**ISSUE:** https://github.com/openedx/platform-roadmap/issues/367

**Dependencies:** None.

**Merge deadline:** None.

Keeping the rest of this template as is for now.

**Reviewers:**
- [ ] tag reviewer
- [ ] tag reviewer

**Merge checklist:**
- [ ] All reviewers approved
- [ ] CI build is green
- [ ] Version bumped
- [ ] Changelog record added
- [ ] Documentation updated (not only docstrings)
- [ ] Commits are squashed

**Post merge:**
- [ ] Create a tag
- [ ] Check new version is pushed to PyPI after tag-triggered build is
      finished.
- [ ] Delete working branch (if not needed anymore)

**Author concerns:** List any concerns about this PR - inelegant
solutions, hacks, quick-and-dirty implementations, concerns about
migrations, etc.

### CODE REVIEW

File: docs/decisions/0016-idv-attempt-events.rst
Comment (MichaelRoytman) [inline docs/decisions/0016-idv-attempt-events.rst:31]: What do you mean by "legacy backend" here?

File: docs/decisions/0016-idv-attempt-events.rst
Comment (MichaelRoytman) [inline docs/decisions/0016-idv-attempt-events.rst:30]: Actually, the intention is for the `verify_student` application to create these events, not any individual IDV plugin. You mention that in the "Consequences section" as well.

These events should be vendor agnostic.

File: openedx_events/learning/data.py
Comment (MichaelRoytman) [inline openedx_events/learning/data.py:607]: Because these events should be vendor agnostic, I think we should remove the `inquiry_id`.

File: openedx_events/learning/signals.py
Comment (MichaelRoytman) [inline openedx_events/learning/signals.py:409]:
```suggestion
# .. event_description: Emitted when a IDV attempt is created
```

I'd recommend keeping it even more generic by removing "by an IDV vendor" from the event descriptions. This is because some forms of IDV are created by humans and not vendors (e.g. `ManualVerification`).

File: openedx_events/learning/data.py
Comment (MichaelRoytman) [inline openedx_events/learning/data.py:600]: This needs to be updated.

File: openedx_events/learning/data.py
Comment (MichaelRoytman) [inline openedx_events/learning/data.py:597]:
```suggestion
    Attributes defined for the Open edX IDV attempt data object.
```

File: docs/decisions/0016-idv-attempt-events.rst
Comment (MichaelRoytman) [inline docs/decisions/0016-idv-attempt-events.rst:42]: Whether these events end up on the event bus depends on the deployment. Without an event bus configuration, these events will be emitted and received as regular Django signals.

Since IDV implementations will be implemented as plugins, there is no cross-service communication that needs to happen over the event bus, even though we have the event bus set up. These events can be handled as normal Django signals.

For that reason, I'd remove the mention of the event bus.

File: docs/decisions/0016-idv-attempt-events.rst
Comment (MichaelRoytman) [inline docs/decisions/0016-idv-attempt-events.rst:1]: Do we need an ADR for this? What I recall from the exams events is that we had a reason for documenting decisions related to those events since they were rather complex and involved a lot of discussion.

These feel like pretty straight forward events, so I'm questioning whether we need to log an entire ADR for it.

How about we don't include the ADR and see what Maria and/or Felipe think when it's time for a review? I'm just trying to save you some work here, so if you'd like to write up the ADR, feel free.

File: openedx_events/learning/data.py
Comment (MichaelRoytman) [inline openedx_events/learning/data.py:604]: `name` and `expiration_date` are both optional, so I think a `default=None` would be sensible here.

**Comment by ilee2u** [comment]
Note for open source reviewers: These events were proposed and approved as part of the Open edX proposal associated with the platform-roadmap ticket in the pull request description.

**Review by MichaelRoytman — CHANGES_REQUESTED** [review]

File: docs/decisions/0016-idv-attempt-events.rst
Comment (ilee2u) [reply docs/decisions/0016-idv-attempt-events.rst:1]: I've added the ADR with the latest revisions for now. Let me think if you think it's short enough that it could be negligible, and I can take it out.

File: openedx_events/learning/data.py
Comment (MichaelRoytman) [inline openedx_events/learning/data.py:603]: "Default to None" should be on the "name" argument, right?

File: openedx_events/learning/data.py
Comment (MichaelRoytman) [inline openedx_events/learning/data.py:600]:
```suggestion
        attempt_id (int): the id of the verification attempt
```
Nit. I think this is a little clearer and doesn't tie the data class to a create or update action.

**Review by MichaelRoytman — APPROVED** [review]
Looks good! I left a few comments I'd like addressed, but I'll approve the pull request. We shouldn't merge this until we get feedback from the owners, though.

**Comment by mariajgrimaldi** [comment]
Thank you, folks, for this contribution. I look forward to seeing it implemented!

**Review by mariajgrimaldi — APPROVED** [review]
I left a minor suggestion. Also, I think one comment from Michael might need your attention. Thank you both for providing the team with such detailed context in the proposal. I'll leave my approval since this implementation aligns with what you proposed and the implementation on the edx-platform https://github.com/openedx/edx-platform/pull/35311.

I'll merge this once all comments are addressed. Thanks!

**Review by mariajgrimaldi — APPROVED** [review]
@ilee2u, thank you. Sorry to bother you but can you please add an entry to the changelog file? Thanks!

**Comment by ilee2u** [comment]
> @ilee2u, thank you. Sorry to bother you but can you please add an entry to the changelog file? Thanks!

@mariajgrimaldi Just added this!

**Pull request merged**
